go/interpreter/lfvm/converter.go

```diff
@@ -21,41 +20,78 @@ import (
 	lru "github.com/hashicorp/golang-lru/v2"
 )
 
-const codeCacheCapacity = 50_000 // up to ~ 1 GB for 22 KiB max code size
+// ConversionConfig contains a set of configuration options for the code conversion.
+type ConversionConfig struct {
+	// CacheSize is the maximum size of the maintained code cache in bytes.
+	// If set to 0, a default size is used. If negative, no cache is used.
+	// Cache sizes are grown in increments of maxCachedCodeLength.
+	// Positive values larger than 0 but less than maxCachedCodeLength are
+	// reported as invalid cache sizes during initialization.
+	CacheSize int
+	// WithSuperInstructions enables the use of super instructions.
+	WithSuperInstructions bool
+}
 
-var cache *lru.Cache[tosca.Hash, Code]
+// Converter converts EVM code to LFVM code.
+type Converter struct {
+	config ConversionConfig
+	cache  *lru.Cache[tosca.Hash, Code]
+}
 
-func init() {
-	res, err := lru.New[tosca.Hash, Code](codeCacheCapacity)
-	if err != nil {
-		panic(fmt.Errorf("failed to create cache: %v", err))
+// NewConverter creates a new code converter with the provided configuration.
+func NewConverter(config ConversionConfig) (*Converter, error) {
+	if config.CacheSize == 0 {
+		config.CacheSize = (1 << 30) // = 1GiB
```

Comment:
nit: if in the future for whatever reason the default size changes, we will have to look for all occurrences of `1<<30`, if it is a constant, it's a simpler change.

Reply:
This is only present exactly here and it is the one place where it is needed and need to be changed. I deliberately did not expose a constant to avoid having code depending in whatever way on it.